rewrite-pattern.js

```diff
@@ -624,6 +624,17 @@ const processTerm = (item, regenerateOptions, groups) => {
 					item = processCharacterClass(item, regenerateOptions, data);
 				}
 			} else if (config.transform.unicodePropertyEscapes) {
+				const caseFoldBMP = configNeedCaseFoldBMP(),
+					  caseFoldUnicode = configNeedCaseFoldUnicode();
+				const singleChars = data.singleChars;
+				if (caseFoldBMP || caseFoldUnicode) {
+					for (const codepoint of singleChars.toArray()) {
```

Comment:
@mathiasbynens It'd be great if regenerate objects were iterable, to avoid having to go through the array :)